airbyte-server/src/main/java/io/airbyte/server/handlers/WorkspacesHandler.java

```diff
@@ -163,6 +164,11 @@ public WorkspaceRead getWorkspaceBySlug(final SlugRequestBody slugRequestBody)
     return buildWorkspaceRead(workspace);
   }
 
+  public WorkspaceRead getWorkspaceByConnectionId(final ConnectionIdRequestBody connectionIdRequestBody) {
+    final StandardWorkspace workspace = configRepository.getStandardWorkspaceFromConnection(connectionIdRequestBody.getConnectionId(), false);
```

Comment:
I think that we should improve that, it is doing a join using 2 queries, we should use a join here.